packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/stackset-types.ts

```diff
@@ -0,0 +1,413 @@
+  /**
+   * Create stack instances in a literal set of accounts or organizational units, and a set of regions
+   *
+   * Stack Instances will be created in every combination of region and account, or region and
+   * Organizational Units (OUs).
+   *
+   * If this is set of Organizational Units, you must have selected `StackSetDeploymentModel.organizations()`
+   * as deployment model.
+   */
+  public static fromList(targets: string[], regions: string[]): StackInstances {
+    if (targets.length === 0) {
+      throw new Error("'targets' may not be an empty list");
+    }
+
+    if (regions.length === 0) {
+      throw new Error("'regions' may not be an empty list");
+    }
+
+    return new class extends StackInstances {
+      public bind(_scope: Construct): StackInstancesBindResult {
+        return {
+          stackSetConfiguration: {
+            DeploymentTargets: targets.join(','),
+            Regions: regions.join(','),
+          },
+        };
+      }
+    }();
+  }
+
+  /**
+   * Create stack instances in a set of accounts or organizational units taken from the pipeline artifacts, and a set of regions
+   *
+   * Stack Instances will be created in every combination of region and account, or region and
+   * Organizational Units (OUs).
+   *
+   * If this is set of Organizational Units, you must have selected `StackSetDeploymentModel.organizations()`
+   * as deployment model.
+   */
+  public static fromArtifact(artifactPath: codepipeline.ArtifactPath, regions: string[]): StackInstances {
+    if (regions.length === 0) {
+      throw new Error("'regions' may not be an empty list");
+    }
+
+    return new class extends StackInstances {
+      public readonly artifactsReferenced?: codepipeline.Artifact[] | undefined = [artifactPath.artifact];
+      public bind(_scope: Construct): StackInstancesBindResult {
+        return {
+          stackSetConfiguration: {
+            DeploymentTargets: artifactPath.location,
+            Regions: regions.join(','),
+          },
+        };
+      }
+    }();
+  }
+
+  /**
+   * The artifacts referenced by the properties of this deployment target
+   *
+   * Does not need to be called by app builders.
+   */
+  readonly artifactsReferenced?: codepipeline.Artifact[];
+
+  /**
+   * Called to attach the stack set instances to a stackset action
+   *
+   * Does not need to be called by app builders.
+   */
+  public abstract bind(scope: Construct): StackInstancesBindResult;
+}
+
+/**
+ * Returned by the StackInstances.bind() function
+ *
+ * Does not need to be used by app builders.
+ */
+export interface StackInstancesBindResult {
+  /**
+   * Properties to mix into the Action configuration
+   */
+  readonly stackSetConfiguration: any;
+}
+
+/**
+ * Describes whether AWS CloudFormation StackSets automatically deploys to AWS Organizations accounts that are added to a target organization or
+ * organizational unit (OU).
+ */
+export enum StackSetOrganizationsAutoDeployment {
```

Comment:
Can we move this closer to where it's first used? Can be even right after that. Right now, this is kind of randomly here, and makes the code a little harder to read because of that.